ui/src/main/java/org/jtaccuino/core/ui/completion/CompletionPopupSkin.java

```diff
@@ -0,0 +1,207 @@
+                    System.out.println("completionList.getParent().getParent().getParent() " + completionList.getParent().getParent().getParent());
+                    completionList.getParent().getParent().minWidth(width);
+                    completionList.getParent().getParent().prefWidth(width);
+                    completionList.getParent().getParent().minWidth(width);
+                    completionList.setMinWidth(width);
+                    completionList.setPrefWidth(width);
+                    completionList.setMaxWidth(width);
+                }
+            }
+        });
+
+        completionList.addEventFilter(KeyEvent.KEY_PRESSED, event -> {
+            int oldSelectedIndex = completionList.getSelectionModel().getSelectedIndex(); // Capture current selection
+            int newCalculatedIndex = oldSelectedIndex; // To store the index after our selection logic
+
+            if (event.getCode() == KeyCode.UP) {
+                completionList.getSelectionModel().selectPrevious();
+                newCalculatedIndex = completionList.getSelectionModel().getSelectedIndex();
+                event.consume();
+            } else if (event.getCode() == KeyCode.DOWN) {
+                completionList.getSelectionModel().selectNext();
+                newCalculatedIndex = completionList.getSelectionModel().getSelectedIndex();
+                event.consume();
+            } else if (event.getCode() == KeyCode.HOME) {
+                completionList.getSelectionModel().selectFirst();
+                newCalculatedIndex = completionList.getSelectionModel().getSelectedIndex();
+                event.consume();
+            } else if (event.getCode() == KeyCode.END) {
+                completionList.getSelectionModel().selectLast();
+                newCalculatedIndex = completionList.getSelectionModel().getSelectedIndex();
+                event.consume();
+            }
+
+            if (newCalculatedIndex != -1) {
+                completionList.getFocusModel().focus(newCalculatedIndex);
+            } else {
+                completionList.getFocusModel().focus(-1);
+            }
+
+            if (newCalculatedIndex != oldSelectedIndex) {
+                final int finalNewCalculatedIndex = newCalculatedIndex; // Needs to be final for lambda
+
+                Platform.runLater(() -> {
+                    completionList.getFocusModel().focus(finalNewCalculatedIndex);
+
+                    completionList.scrollTo(finalNewCalculatedIndex - control.getVisibleCompletions() + 1);
+                });
+            }
+        });
+
+        control.setOnShown((t) -> {
+            Platform.runLater(() -> completionList.requestFocus());
+        });
+    }
+
+    private double calculateListViewPreferredWidth() {
+        if (control.getSuggestions().isEmpty()) {
+            return Region.USE_PREF_SIZE;
+        }
+
+        Text tempText = new Text("M");
```

Comment:
is this correct? `M`?

Reply:
Aah, that is used for the size calculation?
But isn't that more or like a constant and could be determined once?
In that case you could avoid the (repeated) rendering and layout calculation required to give you your "constant" `charWidth`.